### ISSUE TITLE
duplicate keys

### ISSUE THREAD

**Issue by shuther** [AnonymusRaccoon/Kyoo#498]
### Kyoo's version

latest docker

### What happened?

It seems kyoo always scans the full library when I start again the docker?
I have many postgress error about duplicate keys.
1. Is it expected (it is a way for the backend to know if the PK is already added), or still some missing logic?
2. Should I start from a new empty DB?
3. Do you need specific logs if it is worth investigating further (likely within the matcher container?

```
postgres-1     | 2024-05-15 10:13:49.021 UTC [113] ERROR:  duplicate key value violates unique constraint "pk_link_collection_movie"
postgres-1     | 2024-05-15 10:13:49.021 UTC [113] DETAIL:  Key (collection_id, movie_id)=(56f3b9ea-dc65-4db1-8cb2-38555efc2be4, 12965bec-38da-47d6-9340-799105ce5104) already exists.
postgres-1     | 2024-05-15 10:13:49.021 UTC [113] STATEMENT:  INSERT INTO link_collection_movie (collection_id, movie_id)
postgres-1     | 	VALUES ($1, $2)
postgres-1     | 2024-05-15 10:13:49.023 UTC [103] ERROR:  duplicate key value violates unique constraint "pk_link_collection_movie"
```

**Comment by zoriya** [comment]
Thanks for the report!

Duplicated keys are expected, it just means it already exists with the correct values.

On startup, the scanner/matcher will try to add only missing data to kyoo but it sometimes does a bit more than necessary. This is a known thing and I plan on reducing it eventually. Since there is a relatively low amount of data and it's not vital (it's only a few more API calls on startup), this is not a high priority.

I'll keep the issue open for traceability.

**Comment by zoriya** [comment]
Yes, I wanted to do that but my ORM (EF) does not support this yet: https://github.com/dotnet/efcore/issues/16949

**Comment by zoriya** [comment]
Fixed with v5 rework, we properly handle on conflict do update now!
